docs/apm/api.asciidoc

```diff
@@ -553,3 +553,238 @@ The API returns the following:
+For more information, see <<kibana-privileges>>.
+
+[[apm-sourcemap-post-req]]
+===== Request
+
+`POST /api/apm/sourcemaps`
+
+[role="child_attributes"]
+[[apm-sourcemap-post-req-body]]
+===== Request body
+
+`service_name`::
+(required, string) The name of the service that the service map should apply to.
+
+`service_version`::
+(required, string) The version of the service that the service map should apply to.
+
+`bundle_filepath`::
+(required, string) The absolute path of the final bundle as used in the web application.
+
+`sourcemap`::
+(required, string) The source map. It must follow the
+https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k[Source map revision 3 proposal].
+
+[[apm-sourcemap-post-example]]
+===== Example
+
+The following example uploads a sourcemap for a service named `foo` and a service version of `1.0.0`:
+
+[source,curl]
+--------------------------------------------------
+curl -X POST "http://localhost:5601/api/apm/sourcemaps" \
+-H 'Content-Type: multipart/form-data' \
+-H 'kbn-xsrf: true' \
+-H 'Authorization: ApiKey ${YOUR_API_KEY}' \
+--form 'service_name="foo"' \
+--form 'service_version="1.0.0"' \
+--form 'bundle_filepath="/test/e2e/general-usecase/bundle.js.map"' \
+--form 'sourcemap="{\"version\":3,\"file\":\"static/js/main.chunk.js\",\"sources\":[\"fleet-source-map-client/src/index.css\",\"fleet-source-map-client/src/App.js\",\"webpack:///./src/index.css?bb0a\",\"fleet-source-map-client/src/index.js\",\"fleet-source-map-client/src/reportWebVitals.js\"],\"sourcesContent\":[\"content\"],\"mappings\":\"mapping\",\"sourceRoot\":\"\"}"'
+--------------------------------------------------
+
+[[apm-sourcemap-post-body]]
+===== Response body
+
+[source,js]
+--------------------------------------------------
+{
+  "type": "sourcemap",
+  "identifier": "foo-1.0.0",
+  "relative_url": "/api/fleet/artifacts/foo-1.0.0/644fd5a997d1ddd90ee131ba18e2b3d03931d89dd1fe4599143c0b3264b3e456",
+  "body": "eJyFkL1OwzAUhd/Fc+MbYMuCEBIbHRjKgBgc96R16tiWr1OQqr47NwqJxEK3q/PzWccXxchnZ7E1A1SjuhjVZtF2yOxiEPlO17oWox3D3uPFeSRTjmJQARfCPeiAgGx8NTKsYdAc1T3rwaSJGcds8Sp3c1HnhfywUZ3QhMTFFGepZxqMC9oex3CS9tpk1XyozgOlmoVKuJX1DqEQZ0su7PGtLU+V/3JPKc3cL7TJ2FNDRPov4bFta3MDM4f7W69lpJjLO9qdK8bzVPhcJz3HUCQ4LbO/p5hCSC4cZPByrp/wFqOklbpefwAhzpqI",
+  "created": "2021-07-09T20:47:44.812Z",
+  "id": "apm:foo-1.0.0-644fd5a997d1ddd90ee131ba18e2b3d03931d89dd1fe4599143c0b3264b3e456",
+  "compressionAlgorithm": "zlib",
+  "decodedSha256": "644fd5a997d1ddd90ee131ba18e2b3d03931d89dd1fe4599143c0b3264b3e456",
+  "decodedSize": 441,
+  "encodedSha256": "024c72749c3e3dd411b103f7040ae62633558608f480bce4b108cf5b2275bd24",
+  "encodedSize": 237,
+  "encryptionAlgorithm": "none",
+  "packageName": "apm"
+}
+--------------------------------------------------
+
+////
+*******************************************************
+////
+
+[[rum-sourcemap-get]]
+==== Get source maps
+
+Returns an array of Fleet artifacts, including source map uploads.
```

Comment:
I wasn't quite sure how to word what the GET request does. Does this sound accurate? Are there other Fleet artifacts that might be returned by this call?

Reply:
yes, that is correct, you could specify that the source map will be placed inside the `body` property. But they would you the fleet artifact id to delete it.

Reply:
Great, thanks 👍